### PR TITLE
Improve documentation for Style/MixinUsage.

### PR DESCRIPTION
**Pull request by asherkach**
* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [ ] Added tests.
* [ ] Added an entry to the [Changelog](../blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](../blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title
  and description in grammatically correct, complete sentences.
* [x] Run `rake default` or `rake parallel`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by asherkach** [comment]
Is a documentation change considered a user-observable change? If so, I'm happy to add a `CHANGELOG` entry.

File: lib/rubocop/cop/style/mixin_usage.rb [hunk 1]
```diff
@@ -3,11 +3,8 @@
 module RuboCop
   module Cop
     module Style
-      # This cop checks that `include`, `extend` and `prepend` exists at
-      # the top level.
-      # Using these at the top level affects the behavior of `Object`.
-      # There will not be using `include`, `extend` and `prepend` at
-      # the top level. Let's use it inside `class` or `module`.
+      # This cop checks that `include`, `extend` and `prepend` statements appear
+      # inside classes and modules, not at the top level.
```
Comment (koic) [inline lib/rubocop/cop/style/mixin_usage.rb]: Hm. This change seems to make the users unknown about "why" this style is preferred 🤔

Comment (asherkach) [reply lib/rubocop/cop/style/mixin_usage.rb]: That's true of most descriptions, isn't it? IMO, the comment should indicate what the cop does. The style guide should indicate why that style is preferred.

Comment (bbatsov) [reply lib/rubocop/cop/style/mixin_usage.rb]: I don't think this is covered by an entry in the style guide, though.

Comment (asherkach) [reply lib/rubocop/cop/style/mixin_usage.rb]: Fixed to include a description about why that style is preferred.

**Comment by bbatsov** [comment]
I guess we should also add something on the subject in the style guide.

**Pull request merged**